### ISSUE TITLE
[BUG] Invalid Device Ordinal in QUANTILES_BENCH

### ISSUE THREAD

**Issue by dillon-cullinan** [rapidsai/cudf#3392]
**Describe the bug**
The `QUANTILES_BENCH` hits an invalid device ordinal error when ran.

```15:58:50 + cuda-memcheck QUANTILES_BENCH --benchmark_out=QUANTILES_BENCH.json --benchmark_out_format=json
15:58:50 2019-11-13 20:58:50
15:58:50 Running QUANTILES_BENCH
15:58:50 Run on (64 X 3000 MHz CPU s)
15:58:50 CPU Caches:
15:58:50   L1 Data 32K (x32)
15:58:50   L1 Instruction 32K (x32)
15:58:50   L2 Unified 256K (x32)
15:58:50   L3 Unified 40960K (x2)
15:58:50 Load Average: 0.94, 0.80, 0.69
15:58:50 ***WARNING*** CPU scaling is enabled, the benchmark real time measurements may be noisy and will incur extra overhead.
15:58:58 ========= CUDA-MEMCHECK
15:58:58 ========= Program hit cudaErrorInvalidDevice (error 10) due to "invalid device ordinal" on CUDA API call to cudaStreamGetFlags. 
15:58:58 =========     Saved host backtrace up to driver entry point at error
15:58:58 =========     Host Frame:/usr/lib/x86_64-linux-gnu/libcuda.so.1 [0x391b13]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudart.so.10.0 (cudaStreamGetFlags + 0x1a9) [0x383c9]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/librmm.so (cnmemRegisterStream + 0xd1) [0xf3c1]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/librmm.so (_ZN3rmm2mr21cnmem_memory_resource11do_allocateEmP11CUstream_st + 0x152) [0xcd02]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN3rmm5allocIvEE10rmmError_tPPT_mP11CUstream_stPKcj + 0x1b3) [0x367bd3]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN6thrust8cuda_cub11unique_copyINS_6detail22execute_with_allocatorI13rmm_allocatorIcENS0_22execute_on_stream_baseEEENS_17counting_iteratorIiNS_11use_defaultES9_S9_EENS2_15normal_iteratorINS_10device_ptrIiEEEEN70_GLOBAL__N__46_tmpxft_00001a64_00000000_6_sort_helper_cpp1_ii_b4b0973032permuted_row_equality_comparatorILb0EEEEET1_RNS0_16execution_policyIT_EET0_SN_SI_T2_ + 0x1bc) [0x375aac]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN4cudf7groupby4sort6detail6helper13group_offsetsEv + 0x3dc) [0x3724ac]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN4cudf15group_quantilesERKNS_5tableES2_RKSt6vectorIdSaIdEENS_13interpolationEb + 0xb1) [0x46e411]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x1c0e9]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x64c68]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x42351]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x42f2c]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x2a6fd]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x16860]
15:58:58 =========     Host Frame:/lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main + 0xf0) [0x20830]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x168d9]
15:58:58 =========
15:58:58 ========= Program hit cudaErrorInvalidDevice (error 10) due to "invalid device ordinal" on CUDA API call to cudaPeekAtLastError. 
15:58:58 =========     Saved host backtrace up to driver entry point at error
15:58:58 =========     Host Frame:/usr/lib/x86_64-linux-gnu/libcuda.so.1 [0x391b13]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudart.so.10.0 (cudaPeekAtLastError + 0x163) [0x4af33]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN6thrust8cuda_cub11unique_copyINS_6detail22execute_with_allocatorI13rmm_allocatorIcENS0_22execute_on_stream_baseEEENS_17counting_iteratorIiNS_11use_defaultES9_S9_EENS2_15normal_iteratorINS_10device_ptrIiEEEEN70_GLOBAL__N__46_tmpxft_00001a64_00000000_6_sort_helper_cpp1_ii_b4b0973032permuted_row_equality_comparatorILb0EEEEET1_RNS0_16execution_policyIT_EET0_SN_SI_T2_ + 0x519) [0x375e09]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN4cudf7groupby4sort6detail6helper13group_offsetsEv + 0x3dc) [0x3724ac]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN4cudf15group_quantilesERKNS_5tableES2_RKSt6vectorIdSaIdEENS_13interpolationEb + 0xb1) [0x46e411]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x1c0e9]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x64c68]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x42351]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x42f2c]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x2a6fd]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x16860]
15:58:58 =========     Host Frame:/lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main + 0xf0) [0x20830]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x168d9]
15:58:58 =========
15:58:58 ========= Program hit cudaErrorInvalidDevice (error 10) due to "invalid device ordinal" on CUDA API call to cudaPeekAtLastError. 
15:58:58 =========     Saved host backtrace up to driver entry point at error
15:58:58 =========     Host Frame:/usr/lib/x86_64-linux-gnu/libcuda.so.1 [0x391b13]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudart.so.10.0 (cudaPeekAtLastError + 0x163) [0x4af33]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN6thrust8cuda_cub11unique_copyINS_6detail22execute_with_allocatorI13rmmterminate called after throwing an instance of 'thrust::system::system_error'
15:58:58   what():  unique: failed on 2nd step: invalid device ordinal
15:58:58 _allocatorIcENS0_22execute_on_stream_baseEEENS_17counting_iteratorIiNS_11use_defaultES9_S9_EENS2_15normal_iteratorINS_10device_ptrIiEEEEN70_GLOBAL__N__46_tmpxft_00001a64_00000000_6_sort_helper_cpp1_ii_b4b0973032permuted_row_equality_comparatorILb0EEEEET1_RNS0_16execution_policyIT_EET0_SN_SI_T2_ + 0x51e) [0x375e0e]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN4cudf7groupby4sort6detail6helper13group_offsetsEv + 0x3dc) [0x3724ac]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN4cudf15group_quantilesERKNS_5tableES2_RKSt6vectorIdSaIdEENS_13interpolationEb + 0xb1) [0x46e411]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x1c0e9]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x64c68]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x42351]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x42f2c]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x2a6fd]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x16860]
15:58:58 =========     Host Frame:/lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main + 0xf0) [0x20830]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x168d9]
15:58:58 =========
15:58:58 ========= Program hit cudaErrorInvalidDevice (error 10) due to "invalid device ordinal" on CUDA API call to cudaPeekAtLastError. 
15:58:58 =========     Saved host backtrace up to driver entry point at error
15:58:58 =========     Host Frame:/usr/lib/x86_64-linux-gnu/libcuda.so.1 [0x391b13]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudart.so.10.0 (cudaPeekAtLastError + 0x163) [0x4af33]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN6thrust8cuda_cub11unique_copyINS_6detail22execute_with_allocatorI13rmm_allocatorIcENS0_22execute_on_stream_baseEEENS_17counting_iteratorIiNS_11use_defaultES9_S9_EENS2_15normal_iteratorINS_10device_ptrIiEEEEN70_GLOBAL__N__46_tmpxft_00001a64_00000000_6_sort_helper_cpp1_ii_b4b0973032permuted_row_equality_comparatorILb0EEEEET1_RNS0_16execution_policyIT_EET0_SN_SI_T2_ + 0x5dd) [0x375ecd]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN4cudf7groupby4sort6detail6helper13group_offsetsEv + 0x3dc) [0x3724ac]
15:58:58 =========     Host Frame:/conda/envs/cudf_dev/lib/libcudf.so (_ZN4cudf15group_quantilesERKNS_5tableES2_RKSt6vectorIdSaIdEENS_13interpolationEb + 0xb1) [0x46e411]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x1c0e9]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x64c68]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x42351]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x42f2c]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x2a6fd]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x16860]
15:58:58 =========     Host Frame:/lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main + 0xf0) [0x20830]
15:58:58 =========     Host Frame:QUANTILES_BENCH [0x168d9]
15:58:58 =========
15:58:58 ========= Error: process didn't terminate successfully
15:58:58 ========= No CUDA-MEMCHECK results found
15:58:59 Build step 'Execute shell' marked build as failure
```

**Environment overview (please complete the following information)**
 - Environment location: Bare Metal/Docker
 - Method of cuDF install: Source

**Environment details**
Conda env in repo

**Comment by harrism** [comment]
@dillon-cullinan is this still a reproducible issue?

**Comment by karthikeyann** [comment]
I get a different error now. (cuda 10.0)
```
terminate called after throwing an instance of 'thrust::system::system_error'
  what():  unique: failed on 2nd step: unknown error
Aborted (core dumped)
```
This error comes for group_std, group_var also. Bug should be in legacy groupby object.
This is a legacy benchmark `LEGACY_QUANTILES_BENCH`. This benchmark will be retired when legacy APIs are removed. 
Can we make this issue `wontfix` and close?

**Comment by harrism** [comment]
Does the new QUANTILES_BENCH succeed? If so, I think this can be `wontfix`.

**Comment by devavret** [comment]
Just fixed a common dangling pointer error in #4594 . Can someone verify that this fixes the issue?

**Comment by harrism** [comment]
Fixed by #4594.
